refactor(iec): Rename 'Invoke-ExternalCommand()' to 'Start-ExternalProcess()' and support EnvVars/WorkingDirectory

niheaven opened the pull request:
<!-- Provide a general summary of your changes in the Title above -->
<!-- To help with semantic versioning the PR title should start with one of the conventional commit types. -->
<!-- The conventional commit types for Semantic PR are: feat, fix, docs, style, refactor, perf, test, build, ci, chore, revert -->

<!--
  By opening this PR you confirm that you have searched for similar issues/PRs here already.
  Failing to do so will most likely result in closing of this PR without any explanation.
  It is also mandatory to open a relevant issue for discussion with the maintainers,
  before creating any new PR.
  Read the contributing guide first to save both your and our time.
-->

#### Description
<!-- Describe your changes in detail -->

- Rename `Invoke-ExternalCommand()` to `Start-ExternalProcess()`
- Add environment variables and working directory suport

#### Motivation and Context
<!-- Why is this change required? What problem does it solve? -->
<!-- If it fixes an open issue, please link to the issue here. -->

- Rename `Activity` to `Prompt`
- Rename `LogPath` to `LogName`
- Always redirect standard output and error, otherwise the function will not print its output to screen even if `LogName` is not set (related to #5066)
- Allow environment variables be set (related to #5122)
- Allow working directory be set (related to #5122)

#### How Has This Been Tested?
<!-- Please describe in detail how you tested your changes. -->
<!-- Include details of your testing environment, tests ran to see how -->
<!-- your change affects other areas of the code, etc. -->

![image](https://user-images.githubusercontent.com/5832170/186659131-1a65efc7-efc4-4a5d-bfd4-9652b5e63233.png)

With proxy server on and off.

For the strange `git ls-remotes` output: `git` print `From https://github.com/ScoopInstaller/Tests` via the error output 🤣 

#### Checklist:
<!-- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!-- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the [Contributing Guide](https://github.com/ScoopInstaller/.github/blob/main/.github/CONTRIBUTING.md).
- [x] I have ensured that I am targeting the `develop` branch.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.
- [ ] I have added an entry in the CHANGELOG.

r15ch13 commented on lib/core.ps1:
```diff
@@ -474,116 +474,131 @@ function is_local($path) {
+        Out-UTF8File -FilePath $LogName -Append -InputObject $stdoutTask.Result
+        Out-UTF8File -FilePath $LogName -Append -InputObject $stderrTask.Result
+    } else {
+        $stdoutTask.Result | Out-Default
```
```suggestion
        if ($stdoutTask.Result) {
            $stdoutTask.Result.Trim() | Out-Default
        }
```

To prevent the output of a random newline on the console

r15ch13 commented on lib/core.ps1:
```diff
@@ -474,116 +474,131 @@ function is_local($path) {
+        Out-UTF8File -FilePath $LogName -Append -InputObject $stderrTask.Result
+    } else {
+        $stdoutTask.Result | Out-Default
+        $stderrTask.Result | Out-Default
```
```suggestion
        if ($stderrTask.Result) {
            $stderrTask.Result.Trim() | Out-Default
        }
```

niheaven committed:
refactor(iec): Rename to 'Start-ExternalProcess'

- refactor(iec): Rename some parameters
- feat(iec): Support EnvVars in 'Invoke-ExternalCommand()'
- fix(iec): Redirect output when 'CreateNoWindow'
- fix(iec): Allow empty 'ArgumentList'
- feat(iec): Support 'WorkingDirectory'
